src/dll.py

```diff
@@ -40,23 +40,33 @@ def append(self, data):
         new_node._next = None
         new_node._prev = self.tail
         if self.tail:
-            self.tail._next = new_node._prev
+            self.tail._next = new_node
         self.tail = new_node
         if self.head is None:
             self.head = new_node
 
     def pop(self):
         """Remove data from the head of the list and return it."""
         pop_node = self.head
-        self.head._next._prev = None
-        self.head = self.head._next
+        try:
+            self.head._next._prev = None
+            self.head = self.head._next
+        except AttributeError:
+            self.head = None
+            self.tail = None
         return pop_node.data
 
     def shift(self):
         """Remove data from the tail of the list and return it."""
         shift_node = self.tail
-        self.tail._prev._next = None
-        self.tail = self.tail._prev
+        try:
+            print("Self.tail", self.tail.data)
+            self.tail = self.tail._prev
+            self.tail._next = None
+            print("2", self.tail.data)
```

Comment:
same again: make sure to remove debugging print statements from your final submissions.